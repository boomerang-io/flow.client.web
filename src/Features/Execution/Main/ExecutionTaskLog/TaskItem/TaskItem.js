import React from "react";
import PropTypes from "prop-types";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router-dom";
import { Button, ComposedModal, ModalBody } from "@boomerang-io/carbon-addons-boomerang-react";
import ManualTaskModal from "./ManualTaskModal";
import OutputPropertiesLog from "./OutputPropertiesLog";
import TaskApprovalModal from "./TaskApprovalModal";
import TaskExecutionLog from "./TaskExecutionLog";
import moment from "moment";
import dateHelper from "Utils/dateHelper";
import { ApprovalStatus, ExecutionStatus, executionStatusIcon, ExecutionStatusCopy, NodeType } from "Constants";
import styles from "./taskItem.module.scss";

import { appLink } from "Config/appConfig";

const logTaskTypes = ["customtask", "template"];
const logStatusTypes = [ExecutionStatus.Completed, ExecutionStatus.Failure, ExecutionStatus.InProgress];

TaskItem.propTypes = {
  flowActivityId: PropTypes.string.isRequired,
  hidden: PropTypes.bool.isRequired,
  task: PropTypes.object.isRequired,
  executionId: PropTypes.string.isRequired,
};

function TaskItem({ flowActivityId, hidden, task, executionId }) {
  const {
    duration,
    flowTaskStatus,
    id,
    outputs,
    startTime,
    taskId,
    taskName,
    approval,
    taskType,
    switchValue,
    runWorkflowActivityId,
    runWorkflowId,
    runWorkflowActivityStatus,
  } = task;
  // const Icon = executionStatusIcon[flowTaskStatus];
  // const statusClassName = styles[flowTaskStatus];
  let statusClassName;
  let Icon;
  let runStatus;
  if (taskType === NodeType.RunWorkflow) {
    statusClassName = styles[runWorkflowActivityStatus] ?? styles[flowTaskStatus];
    Icon = executionStatusIcon[runWorkflowActivityStatus] ?? executionStatusIcon[flowTaskStatus];
    runStatus = runWorkflowActivityStatus ?? flowTaskStatus;
  } else {
    statusClassName = styles[flowTaskStatus];
    Icon = executionStatusIcon[flowTaskStatus];
    runStatus = flowTaskStatus;
  }

  const calculatedDuration = Number.parseInt(duration)
    ? dateHelper.timeMillisecondsToTimeUnit(duration)
    : dateHelper.durationFromThenToNow(startTime) || "---";

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
    <li key={id} id={`task-${taskId}`} tabIndex={0} className={`${styles.taskitem} ${statusClassName}`}>
      <div className={styles.progressBar} />
      <section className={styles.header}>
        <div className={styles.title}>
          <Icon aria-label={runStatus} className={styles.taskIcon} />
          <p title={taskName} data-testid="taskitem-name">
            {taskName}
          </p>
        </div>
        <div className={`${styles.status} ${statusClassName}`}>
          <Icon aria-label={runStatus} className={styles.statusIcon} />
          <p>{ExecutionStatusCopy[runStatus]}</p>
        </div>
      </section>
      <section className={styles.data}>
        <div className={styles.time}>
          <p className={styles.timeTitle}>Start time</p>
          <time className={styles.timeValue}>{startTime ? moment(startTime).format("hh:mm:ss A") : "---"}</time>
        </div>
        <div className={styles.time}>
          <p className={styles.timeTitle}>Duration</p>
          <time className={styles.timeValue}>{calculatedDuration}</time>
        </div>
        {taskType === NodeType.Decision && (
          <div className={styles.time}>
            <p className={styles.timeTitle}>Value</p>
            <time className={styles.timeValue}>{switchValue ?? ""}</time>
          </div>
        )}
      </section>
      {!hidden && (
        <section className={styles.data}>
          {((flowTaskStatus === ExecutionStatus.Cancelled && duration > 0) || (logTaskTypes.includes(taskType) && logStatusTypes.includes(runStatus))) && (
            <TaskExecutionLog flowActivityId={flowActivityId} flowTaskId={taskId} flowTaskName={taskName} />
          )}
          {outputs && Object.keys(outputs).length > 0 && (
            <OutputPropertiesLog flowTaskName={taskName} flowTaskOutputs={outputs} />
          )}
          {taskType === NodeType.RunWorkflow && runWorkflowActivityId && runWorkflowId && (
            <Link
              to={appLink.execution({ executionId: runWorkflowActivityId, workflowId: runWorkflowId })}
              className={styles.viewActivityLink}
            >
              View Activity
            </Link>
          )}
          {flowTaskStatus !== ExecutionStatus.Cancelled && taskType === NodeType.Approval && approval?.status === ApprovalStatus.Submitted && (
            <ComposedModal
              modalHeaderProps={{
                title: "Action Manual Approval",
                subtitle: taskName,
              }}
              modalTrigger={({ openModal }) => (
                <Button className={styles.modalTrigger} size="small" kind="ghost" onClick={openModal}>
                  Action Manual Approval
                </Button>
              )}
            >
              {({ closeModal }) => (
                <TaskApprovalModal
                  approvalId={approval.id}
                  flowTaskName={taskName}
                  executionId={executionId}
                  closeModal={closeModal}
                />
              )}
            </ComposedModal>
          )}
          {flowTaskStatus !== ExecutionStatus.Cancelled && taskType === NodeType.Manual && approval?.status === ApprovalStatus.Submitted && (
            <ComposedModal
              composedModalProps={{
                containerClassName: styles.actionManualTaskModalContainer,
              }}
              modalHeaderProps={{
                title: "Action Manual Task",
                subtitle: taskName,
              }}
              modalTrigger={({ openModal }) => (
                <Button className={styles.modalTrigger} size="small" kind="ghost" onClick={openModal}>
                  Action Manual Task
                </Button>
              )}
            >
              {({ closeModal }) => (
                <ManualTaskModal
                  approvalId={approval?.id}
                  flowTaskName={taskName}
                  executionId={executionId}
                  closeModal={closeModal}
                  instructions={approval?.instructions}
                />
              )}
            </ComposedModal>
          )}
          {taskType === NodeType.Approval &&
            (approval?.status === ApprovalStatus.Approved || approval?.status === ApprovalStatus.Rejected) && (
              <ComposedModal
                composedModalProps={{
                  containerClassName: styles.approvalResultsModalContainer,
                  shouldCloseOnOverlayClick: true,
                }}
                modalHeaderProps={{
                  title: "Manual Approval details",
                }}
                modalTrigger={({ openModal }) => (
                  <Button className={styles.modalTrigger} size="small" kind="ghost" onClick={openModal}>
                    View Manual Approval
                  </Button>
                )}
              >
                {() => (
                  <ModalBody>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval Status</span>
                      <p className={styles.sectionDetail}>{approval.status}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approver</span>
                      <p
                        className={styles.sectionDetail}
                      >{`${approval.audit.approverName} (${approval.audit.approverEmail})`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval submitted</span>
                      <p className={styles.sectionDetail}>
                        {moment(approval.audit.actionDate).format("YYYY-MM-DD hh:mm A")}
                      </p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval comments</span>
                      <p className={styles.sectionDetail}>{approval.audit.comments}</p>
                    </section>
                  </ModalBody>
                )}
              </ComposedModal>
            )}
          {taskType === NodeType.Manual &&
            (approval?.status === ApprovalStatus.Approved || approval?.status === ApprovalStatus.Rejected) && (
              <ComposedModal
                composedModalProps={{
                  containerClassName: styles.approvalResultsModalContainer,
                  shouldCloseOnOverlayClick: true,
                }}
                modalHeaderProps={{
                  title: "Manual Task details",
                }}
                modalTrigger={({ openModal }) => (
                  <Button className={styles.modalTrigger} size="small" kind="ghost" onClick={openModal}>
                    View Manual Task
                  </Button>
                )}
              >
                {() => (
                  <ModalBody>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Status</span>
                      <p className={styles.sectionDetail}>{`${
                        approval.status === ApprovalStatus.Approved
                          ? "Successfully Completed"
                          : "Unsuccessfully Completed"
                      }`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approver</span>
                      <p
                        className={styles.sectionDetail}
                      >{`${approval.audit.approverName} (${approval.audit.approverEmail})`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Submitted</span>
                      <p className={styles.sectionDetail}>
                        {moment(approval.audit.actionDate).format("YYYY-MM-DD hh:mm A")}
                      </p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Instructions</span>
                      <ReactMarkdown className="markdown-body" source={approval?.instructions} />
                    </section>
                  </ModalBody>
                )}
              </ComposedModal>
            )}
        </section>
      )}
    </li>
  );
}

export default TaskItem;
