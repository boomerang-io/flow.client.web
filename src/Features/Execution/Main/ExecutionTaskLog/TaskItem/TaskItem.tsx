import React from "react";
import ReactMarkdown from "react-markdown";
import { Link, useParams } from "react-router-dom";
import { Button, ModalBody } from "@carbon/react";
import { ComposedModal } from "@boomerang-io/carbon-addons-boomerang-react";
import ErrorModal from "Components/ErrorModal";
import ManualTaskModal from "./ManualTaskModal";
import OutputPropertiesLog from "./OutputPropertiesLog";
import TaskApprovalModal from "./TaskApprovalModal";
import TaskExecutionLog from "./TaskExecutionLog";
import moment from "moment";
import dateHelper from "Utils/dateHelper";
import { executionStatusIcon, ExecutionStatusCopy, NodeType } from "Constants";
import { ApprovalStatus, RunStatus, WorkflowExecutionStep } from "Types";
import styles from "./taskItem.module.scss";

import { appLink } from "Config/appConfig";

const logTaskTypes = ["customtask", "template", "script"];
const logStatusTypes = [RunStatus.Succeeded, RunStatus.Failed, RunStatus.Running];

type Props = {
  flowActivityId: string;
  hidden: boolean;
  task: WorkflowExecutionStep;
  executionId: string;
};

function TaskItem({ flowActivityId, hidden, task, executionId }: Props) {
  const { team } = useParams<{ team: string }>();
  const {
    approval,
    duration,
    flowTaskStatus,
    id,
    results,
    runWorkflowActivityId,
    runWorkflowActivityStatus,
    runWorkflowId,
    startTime,
    taskId,
    taskName,
    taskType,
    switchValue,
    error,
  } = task;
  // const Icon = executionStatusIcon[flowTaskStatus];
  // const statusClassName = styles[flowTaskStatus];
  let statusClassName;
  let Icon;
  let runStatus;
  if (taskType === NodeType.RunWorkflow) {
    statusClassName = styles[runWorkflowActivityStatus] ?? styles[flowTaskStatus];
    Icon = executionStatusIcon[runWorkflowActivityStatus] ?? executionStatusIcon[flowTaskStatus];
    runStatus = runWorkflowActivityStatus ?? flowTaskStatus;
  } else {
    statusClassName = styles[flowTaskStatus];
    Icon = executionStatusIcon[flowTaskStatus];
    runStatus = flowTaskStatus;
  }
  //@ts-ignore
  const calculatedDuration = Number.parseInt(duration)
    ? dateHelper.timeMillisecondsToTimeUnit(duration)
    : dateHelper.durationFromThenToNow(startTime) || "---";

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
    <li key={id} id={`task-${taskId}`} tabIndex={0} className={`${styles.taskitem} ${statusClassName}`}>
      <div className={styles.progressBar} />
      <section className={styles.header}>
        <div className={styles.title}>
          <Icon aria-label={runStatus} className={styles.taskIcon} />
          <p title={taskName} data-testid="taskitem-name">
            {taskName}
          </p>
        </div>
        <div className={`${styles.status} ${statusClassName}`}>
          <Icon aria-label={runStatus} className={styles.statusIcon} />
          <p>{ExecutionStatusCopy[runStatus]}</p>
        </div>
      </section>
      <section className={styles.data}>
        <div className={styles.time}>
          <p className={styles.timeTitle}>Start time</p>
          <time className={styles.timeValue}>{startTime ? moment(startTime).format("hh:mm:ss A") : "---"}</time>
        </div>
        <div className={styles.time}>
          <p className={styles.timeTitle}>Duration</p>
          <time className={styles.timeValue}>{calculatedDuration}</time>
        </div>
        {taskType === NodeType.Decision && (
          <div className={styles.time}>
            <p className={styles.timeTitle}>Value</p>
            <time className={styles.timeValue}>{switchValue ?? ""}</time>
          </div>
        )}
      </section>
      {!hidden && (
        <section className={styles.data}>
          {((flowTaskStatus === RunStatus.Cancelled && duration > 0) ||
            (logTaskTypes.includes(taskType) && logStatusTypes.includes(runStatus))) && (
            <TaskExecutionLog flowActivityId={flowActivityId} flowTaskId={taskId} flowTaskName={taskName} />
          )}
          {results && Object.keys(results).length > 0 && (
            //@ts-ignore
            <OutputPropertiesLog flowTaskName={taskName} flowTaskOutputs={results} />
          )}
          {taskType === NodeType.RunWorkflow && runWorkflowActivityId && runWorkflowId && (
            <Link
              to={appLink.execution({ team, executionId: runWorkflowActivityId, workflowId: runWorkflowId })}
              className={styles.viewActivityLink}
            >
              View Activity
            </Link>
          )}
          {Boolean(error?.code) && (
            <ComposedModal
              modalHeaderProps={{
                title: "View Task Error",
                subtitle: taskName,
              }}
              modalTrigger={({ openModal }: { openModal: () => void }) => (
                <Button size="sm" kind="ghost" onClick={openModal}>
                  View Error
                </Button>
              )}
            >
              {({ closeModal }: { closeModal: () => void }) => (
                <ErrorModal errorCode={error?.code ?? ""} errorMessage={error?.message ?? ""} />
              )}
            </ComposedModal>
          )}
          {flowTaskStatus !== RunStatus.Cancelled &&
            taskType === NodeType.Approval &&
            approval?.status === ApprovalStatus.Submitted && (
              <ComposedModal
                modalHeaderProps={{
                  title: "Action Manual Approval",
                  subtitle: taskName,
                }}
                modalTrigger={({ openModal }: { openModal: () => void }) => (
                  <Button className={styles.modalTrigger} size="sm" kind="ghost" onClick={openModal}>
                    Action Manual Approval
                  </Button>
                )}
              >
                {({ closeModal }: { closeModal: () => void }) => (
                  <TaskApprovalModal approvalId={approval.id} executionId={executionId} closeModal={closeModal} />
                )}
              </ComposedModal>
            )}
          {flowTaskStatus !== RunStatus.Cancelled &&
            taskType === NodeType.Manual &&
            approval?.status === ApprovalStatus.Submitted && (
              <ComposedModal
                composedModalProps={{
                  containerClassName: styles.actionManualTaskModalContainer,
                }}
                modalHeaderProps={{
                  title: "Action Manual Task",
                  subtitle: taskName,
                }}
                modalTrigger={({ openModal }: { openModal: () => void }) => (
                  <Button className={styles.modalTrigger} size="sm" kind="ghost" onClick={openModal}>
                    Action Manual Task
                  </Button>
                )}
              >
                {({ closeModal }: { closeModal: () => void }) => (
                  <ManualTaskModal
                    approvalId={approval?.id}
                    executionId={executionId}
                    closeModal={closeModal}
                    instructions={approval?.instructions}
                  />
                )}
              </ComposedModal>
            )}
          {taskType === NodeType.Approval &&
            (approval?.status === ApprovalStatus.Approved || approval?.status === ApprovalStatus.Rejected) && (
              <ComposedModal
                composedModalProps={{
                  containerClassName: styles.approvalResultsModalContainer,
                  shouldCloseOnOverlayClick: true,
                }}
                modalHeaderProps={{
                  title: "Manual Approval details",
                }}
                modalTrigger={({ openModal }: { openModal: () => void }) => (
                  <Button className={styles.modalTrigger} size="sm" kind="ghost" onClick={openModal}>
                    View Manual Approval
                  </Button>
                )}
              >
                {() => (
                  <ModalBody>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval Status</span>
                      <p className={styles.sectionDetail}>{approval.status}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approver</span>
                      <p
                        className={styles.sectionDetail}
                      >{`${approval.audit.approverName} (${approval.audit.approverEmail})`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval submitted</span>
                      <p className={styles.sectionDetail}>
                        {moment(approval.audit.actionDate).format("YYYY-MM-DD hh:mm A")}
                      </p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approval comments</span>
                      <p className={styles.sectionDetail}>{approval.audit.comments}</p>
                    </section>
                  </ModalBody>
                )}
              </ComposedModal>
            )}
          {taskType === NodeType.Manual &&
            (approval?.status === ApprovalStatus.Approved || approval?.status === ApprovalStatus.Rejected) && (
              <ComposedModal
                composedModalProps={{
                  containerClassName: styles.approvalResultsModalContainer,
                  shouldCloseOnOverlayClick: true,
                }}
                modalHeaderProps={{
                  title: "Manual Task details",
                }}
                modalTrigger={({ openModal }: { openModal: () => void }) => (
                  <Button className={styles.modalTrigger} size="sm" kind="ghost" onClick={openModal}>
                    View Manual Task
                  </Button>
                )}
              >
                {() => (
                  <ModalBody>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Status</span>
                      <p className={styles.sectionDetail}>{`${
                        approval.status === ApprovalStatus.Approved
                          ? "Successfully Completed"
                          : "Unsuccessfully Completed"
                      }`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Approver</span>
                      <p
                        className={styles.sectionDetail}
                      >{`${approval.audit.approverName} (${approval.audit.approverEmail})`}</p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Submitted</span>
                      <p className={styles.sectionDetail}>
                        {moment(approval.audit.actionDate).format("YYYY-MM-DD hh:mm A")}
                      </p>
                    </section>
                    <section className={styles.detailedSection}>
                      <span className={styles.sectionHeader}>Instructions</span>
                      <ReactMarkdown className="markdown-body" children={approval?.instructions} />
                    </section>
                  </ModalBody>
                )}
              </ComposedModal>
            )}
        </section>
      )}
    </li>
  );
}

export default TaskItem;
