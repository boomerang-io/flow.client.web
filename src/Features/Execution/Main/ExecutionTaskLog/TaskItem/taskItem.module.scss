@import "Styles/mixins";

.taskitem {
  height: 12.5rem;
  position: relative;
  width: 100%;

  &:focus {
    @include focus-outline(var(--cds-interactive-01), -0.125rem);
  }

  &.completed {
    border-top: 0.25rem solid var(--cds-support-02);
  }

  &.waiting {
    // border-top: 0.25rem solid var(--cds-support-03);
    border-top: 0.25rem solid var(--flow-wait);
  }

  &.failure {
    border-top: 0.25rem solid var(--flow-failure);
  }

  &.inProgress {
    border-top: 0.25rem solid #d5d9e0;

    .progressBar {
      position: absolute;
      display: block;
      height: 0.25rem;
      top: -0.25rem;
      left: 0;
      transition: width 0.3s linear;
      width: 100%;
      overflow: hidden;

      &:before {
        content: "";
        position: absolute;
        background-color: var(--cds-interactive-01);
        top: 0;
        left: 0;
        bottom: 0;
        will-change: left, right;
        animation: indeterminate 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;
      }

      &:after {
        content: "";
        position: absolute;
        background-color: var(--cds-interactive-01);
        top: 0;
        left: 0;
        bottom: 0;
        will-change: left, right;
        animation: indeterminate-short 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;
        animation-delay: 1.15s;
      }
    }
  }

  &.notstarted,
  &.invalid,
  &.skipped {
    border-top: 0.25rem solid #d5d9e0;
  }

  &:last-of-type {
    border-bottom: 0.0625rem solid #d5d9e0;
  }
}

.header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 1.25rem 1rem 0.25rem 1rem;
}

.title {
  align-items: center;
  color: var(--cds-text-01);
  display: flex;
  font-size: 0.875rem;
  line-height: 1rem;
  white-space: nowrap;

  p {
    max-width: 14rem;
    @include text-overflow-ellipsis();
  }
}

.taskIcon {
  height: 1rem;
  margin-right: 0.5rem;
  width: 1rem;
}

.status {
  align-items: center;
  display: flex;
  font-size: 0.875rem;
  letter-spacing: 0.01rem;
  line-height: 1rem;
  white-space: nowrap;

  &.completed {
    color: var(--cds-support-02);
    svg {
      fill: var(--cds-support-02);
    }
  }

  &.waiting {
    // color: var(--cds-support-03);
    color: var(--flow-wait);

    svg {
      // fill: var(--cds-support-03);
      fill: var(--flow-wait);
    }
  }

  &.failure {
    color: var(--flow-failure);
    svg {
      fill: var(--flow-failure);
    }
  }

  &.inProgress,
  &.notstarted,
  &.invalid,
  &.skipped {
    color: #6f7878;
    svg {
      fill: #6f7878;
    }
  }
}

.statusIcon {
  height: 1rem;
  margin-right: 0.5rem;
  width: 1rem;
}

.data {
  margin-left: 2.5rem;
  margin-top: 2rem;
  white-space: nowrap;
}

.modalTrigger {
  margin-left: -0.75rem;
}

.subtitle {
  color: var(--cds-text-01);
  font-size: 0.75rem;
  line-height: 0.9375rem;
  margin-bottom: 1.5rem;
}

.time {
  display: flex;
  margin-bottom: 1rem;

  &:last-of-type {
    margin-bottom: 1.5rem;
  }
}

.timeTitle {
  color: var(--cds-text-01);
  font-size: 0.875rem;
  font-weight: 300;
  line-height: 1.125rem;
  width: 4.8125rem;
}

.timeValue {
  color: var(--cds-text-01);
  font-size: 0.875rem;
  line-height: 1.125rem;
}

.approvalResultsModalContainer {
  max-height: 40rem;
  max-width: 75ch;
}

.actionManualTaskModalContainer {
  max-height: 40rem;
  max-width: 75ch;
}

.detailedSection {
  margin-bottom: 1.5rem;
}

.sectionHeader {
  color: #4d5358;
  font-size: 0.75rem;
  letter-spacing: 0.00875rem;
  line-height: 0.9375rem;
  margin-bottom: 0.5rem;
}

.sectionDetail {
  color: var(--cds-text-01);
  font-size: 0.875rem;
  letter-spacing: 0.01rem;
  line-height: 1.125rem;
}

.viewActivityLink {
  color: var(--cds-interactive-01);
  font-size: 0.8rem;
}

@keyframes indeterminate {
  0% {
    left: -35%;
    right: 100%;
  }
  60% {
    left: 100%;
    right: -90%;
  }
  100% {
    left: 100%;
    right: -90%;
  }
}

@keyframes indeterminate-short {
  0% {
    left: -200%;
    right: 100%;
  }
  60% {
    left: 107%;
    right: -8%;
  }
  100% {
    left: 107%;
    right: -8%;
  }
}
