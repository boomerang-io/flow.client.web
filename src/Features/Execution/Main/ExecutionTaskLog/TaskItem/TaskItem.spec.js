import React from "react";
import TaskItem from "./index";

const props = {
  flowActivityId: "1",
  task: {
    activityId: "5c36289096052900012cc81d",
    duration: 300190,
    flowTaskStatus: "completed",
    id: "5c36289096052900012cc81e",
    order: 1,
    startTime: "2019-09-03T15:00:00.230+0000",
    taskId: "515c8b05-ceb0-470a-a58e-b8740b332a6a",
    outputs: {
      args: {
        test: "true",
      },
    },
    taskName: "Send Slack Message",
  },
  hidden: false,
};

describe("TaskItem --- Snapshot", () => {
  it("Capturing Snapshot of TaskItem", () => {
    const { baseElement } = rtlRender(<TaskItem {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
