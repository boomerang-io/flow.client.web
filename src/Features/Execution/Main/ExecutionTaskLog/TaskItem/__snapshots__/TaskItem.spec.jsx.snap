// Vitest Snapshot v1

exports[`TaskItem --- Snapshot > Capturing Snapshot of TaskItem 1`] = `
<body
  id="app"
>
  <div>
    <li
      class="undefined undefined"
      id="task-515c8b05-ceb0-470a-a58e-b8740b332a6a"
      tabindex="0"
    >
      <div />
      <section>
        <div>
          <svg
            aria-label="completed"
            fill="currentColor"
            focusable="false"
            height="32"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 21.414L9 16.413 10.413 15 14 18.586 21.585 11 23 12.415 14 21.414z"
            />
            <path
              d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
            />
          </svg>
          <p
            data-testid="taskitem-name"
            title="Send Slack Message"
          >
            Send Slack Message
          </p>
        </div>
        <div
          class="undefined undefined"
        >
          <svg
            aria-label="completed"
            fill="currentColor"
            focusable="false"
            height="32"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 21.414L9 16.413 10.413 15 14 18.586 21.585 11 23 12.415 14 21.414z"
            />
            <path
              d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
            />
          </svg>
          <p>
            Succeeded
          </p>
        </div>
      </section>
      <section>
        <div>
          <p>
            Start time
          </p>
          <time>
            03:00:00 PM
          </time>
        </div>
        <div>
          <p>
            Duration
          </p>
          <time>
            5 mins
          </time>
        </div>
      </section>
      <section />
    </li>
  </div>
</body>
`;
